### DISCUSSION TITLE
Assigning shell value to environment variable?

### DISCUSSION THREAD

**Discussion by ajmas** [community/community#27199] [Actions]
<p>I am needing to set an environment variable based on an execution of shell script, but I haven’t been able to get it to work.</p>
<pre><code class="lang-auto">      - name: Setup versions
        run: export VERSION=99.99.99; DATE=`date +%Y%m%d`
</code></pre>
<p>values aren’t listed in the <code>env</code> in the next step.</p>
<p>I did also try:</p>
<pre><code class="lang-auto">env:
  VERSION: 99.99.99
   DATE: `date +%Y%m%d`
</code></pre>
<p>but that is invalid syntax per YAML.</p>
<p>Any suggestions?</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>Each <code>run</code> step gets its own shell instance, so <code>export</code>ed variables don’t carry over. If you want that, you need to write to the <code>GITHUB_ENV</code> environment file, e.g.:</p>
<pre><code class="lang-yaml">- name: Setup versions
  run: |
    echo "VERSION=99.99.99" &gt;&gt;${GITHUB_ENV}
    echo "DATE=$(date +%Y%m%d)" &gt;&gt;${GITHUB_ENV}
</code></pre>
<p>The <code>VERSION</code> and <code>DATE</code> environment variables will be available in all subsequent steps of the job.</p>
<p>For details, see <a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-commands-for-github-actions#setting-an-environment-variable">Setting an environment variable</a>.</p>
